packages/apps/human-app/server/src/modules/user-worker/worker.service.ts

```diff
@@ -1,16 +1,31 @@
 import { Injectable } from '@nestjs/common';
 import { ReputationOracleGateway } from '../../integrations/reputation-oracle/reputation-oracle.gateway';
-import { SignupWorkerCommand } from './model/worker-registration.model';
+import { ExchangeOracleGateway } from '../../integrations/exchange-oracle/exchange-oracle.gateway';
+import {
+  RegisterWorkerCommand,
+  SignupWorkerCommand,
+} from './model/worker-registration.model';
 import { SigninWorkerCommand } from './model/worker-signin.model';
 
 @Injectable()
 export class WorkerService {
-  constructor(private gateway: ReputationOracleGateway) {}
+  constructor(
+    private exchangeOracleGateway: ExchangeOracleGateway,
+    private reputationOracleGateway: ReputationOracleGateway,
+  ) {}
 
   async signupWorker(signupWorkerCommand: SignupWorkerCommand) {
-    return this.gateway.sendWorkerSignup(signupWorkerCommand);
+    return this.reputationOracleGateway.sendWorkerSignup(signupWorkerCommand);
   }
   async signinWorker(signinWorkerCommand: SigninWorkerCommand) {
-    return this.gateway.sendWorkerSignin(signinWorkerCommand);
+    return this.reputationOracleGateway.sendWorkerSignin(signinWorkerCommand);
+  }
+  async registerWorker(registerWorkerCommand: RegisterWorkerCommand) {
+    // TODO: Check if the oracle has registration_needed: true in KVStore else throw an error
```

Comment:
Add this

Reply:
**Eugene**
Another point, which chain is this oracle_address registered for? ChainId will be needed to request information in KVStore.

**Sergey Dzeranov**
To know that the user is registered in the exchange oracle, if registration is required there. And what does kvstore have to do with this?

**Eugene**
Where do we request this parameter? registration_needed: true Or will this check work directly on the oracle?

**Sergey Dzeranov**
For some exchange oracles (most likely for all except fortune), additional registration of the user on some platform (like cvat) will be required. And we need to store information that the user has completed this registration so that we don't show instructions for registration for a specific oracle on the front end every time.
Are you talking about an additional check on the human app backend before sending a request to kvstore?
You can comment out this check for now, and we will discuss it when Adrian returns, as he created the task.
In theory, the information about chainId can be passed from the frontend.

**Eugene**
Yes, before sending a request to the exchange oracle for registration. If this variable in KVStore is false, there will be no point in sending the registration request, as I understand it.
In theory, the information about chainId can be passed from the frontend.
Let's leave it without the check for now, I'll write a TODO.

Reply:
Let's just use by default Polygon for mainnet and Amoy for testnet